src/main/java/org/opensearch/security/action/onbehalf/CreateOnBehalfOfTokenAction.java

```diff
@@ -152,8 +152,8 @@ public void accept(RestChannel channel) throws Exception {
                         user.getName(),
                         service,
                         tokenDuration,
-                        mappedRoles.stream().collect(Collectors.toList()),
-                        user.getRoles().stream().collect(Collectors.toList()),
+                        mappedRoles.stream().collect(Collectors.toSet()),
```

Comment:
Had to swap to set in order to avoid needing to change other classes more extensively.

Reply:
You can use `Set.copyOf()` instead of  `.strem().collect()`

Reply:
Will do!